### DISCUSSION TITLE
Is there a syntax highlighting textbox for Avalonia?

### DISCUSSION THREAD

**Discussion by alexandrehtrb** [AvaloniaUI/Avalonia#8851] [Q&A] [answered]
Hello,

Is there a TextBox with syntax highlighting available for Avalonia?
I searched this before and found RichTextBox, but it was not ported at that time... will it be ported to Avalonia?
For example, a JSON like in the image below, but with coloured text:

![image](https://user-images.githubusercontent.com/27026741/187239522-1554caf7-8cda-4234-8a77-206b7d773878.png)

**Comment by Gillibald** [comment, marked as answer]
Have a look at:
https://github.com/AvaloniaUI/AvaloniaEdit
https://www.nuget.org/packages/Avalonia.AvaloniaEdit/11.0.0-preview1

**Reply by timunie** [reply]
You were a second faster 🤣

**Comment by timunie** [comment]
For that use case consider AvaloniaEdit https://github.com/AvaloniaUI/AvaloniaEdit. 

RichTextBox is hard to implement. If and when it will arrive is written in the ✨. The team wants to add it, but yeah its not easy. 

Happy coding
Tim

**Reply by rabbitism** [reply]
I have to say, even textbox it self is hard to implement. I tried to restyle textbox yesterday, and I found there are so many details to handle!
